UnityProject/Assets/Scripts/Util/GameObjectExtensions.cs

```diff
@@ -165,4 +167,67 @@ public static bool IsAtHiddenPos(this GameObject gameObject)
 		// be considered hidden (nothing in the game should be this low except hidden stuff)
 		return gameObject.transform.position.z <= (TransformState.HiddenPos.z + 10);
 	}
+
+	/// <summary>
+	/// Removes all children of this given GameObject.
+	/// </summary>
+	public static void DeleteAllChildren(this GameObject gameObject)
+	{
+		for (int i = 0; i < gameObject.transform.childCount; i++)
+		{
+			Object.Destroy(gameObject.transform.GetChild(i).gameObject);
+		}
+	}
+
+	/// <summary>
+	/// Finds all items/gameObjects with specific a component that are nearby a target that aren't easily detected using physics functions.
+	/// WARNING: This process is extremely slow, and goes over multiple frames (Can take up to 0.1 or 3 seconds depending on how large the scene is).
+	/// This function will never stall the game/server unless you repeatedly call coroutine multiple times and end up creating a lot of GC.
+	/// </summary>
+	/// <param name="maximumDistance">The maximum distance between the component and the target (1 Tile is 1.25)</param>
+	/// <param name="target">The target gameObject, usually the player.</param>
+	/// <param name="result">What to do with this information once its all been processed?</param>
+	/// <param name="matrix">What matrix do you want to scan?</param>
+	/// <typeparam name="T">The component you're looking for</typeparam>
+	/// <returns>A list of components nearby components to target.</returns>
+	public static IEnumerator FindAllComponentsNearestToTarget<T>(this GameObject target, float maximumDistance, System.Action<List<T>> result, MatrixInfo matrix = null)
+	{
+		if (result == null)
+		{
+			Loggy.LogError("[MatrixManager/FindAllComponentsNearestToTarget()] - Cannot start coroutine without a result effect.");
+			yield break;
+		}
+#if UNITY_EDITOR
+		var stopwatch = new Stopwatch();
+		stopwatch.Start();
+#endif
+		if (matrix == null) matrix = MatrixManager.MainStationMatrix;
+		var currentIndex = 0;
+		var maximumIndexes = 20;
```

Comment:
might be worth doing one frame since it doesn't matter too much if it's a bit slow?

Reply:
replaced it with another method